frontend/src/components/widgets/DataFrame/columns/utils.ts

```diff
@@ -0,0 +1,358 @@
+  return {
+    id: column.id,
+    title: column.title,
+    hasMenu: false,
+    themeOverride: column.themeOverride,
+    ...(column.isStretched && {
+      grow: column.isIndex ? 1 : 3,
+    }),
+    ...(column.width && {
+      width: column.width,
+    }),
+  } as GridColumn
+}
+
+/**
+ * Merges the default column parameters with the user-defined column parameters.
+ *
+ * @param defaultParams - The default column parameters.
+ * @param userParams - The user-defined column parameters.
+ *
+ * @returns The merged column parameters.
+ */
+export function mergeColumnParameters(
+  defaultParams: Record<string, any> | undefined | null,
+  userParams: Record<string, any> | undefined | null
+): Record<string, any> {
+  if (isNullOrUndefined(defaultParams)) {
+    return userParams || {}
+  }
+
+  if (isNullOrUndefined(userParams)) {
+    return defaultParams || {}
+  }
+
+  return merge(defaultParams, userParams)
+}
+
+/**
+ * Converts the given value of unknown type to an array without
+ * the risks of any exceptions.
+ *
+ * @param data - The value to convert to an array.
+ *
+ * @returns The converted array or an empty array if the value cannot be interpreted as an array.
+ */
+export function toSafeArray(data: any): any[] {
+  if (isNullOrUndefined(data)) {
+    return []
+  }
+
+  if (typeof data === "number" || typeof data === "boolean") {
+    // Single number or boolean
+    return [data]
+  }
+
+  if (typeof data === "string") {
+    if (data === "") {
+      // Empty string
+      return []
+    }
+
+    // Try to parse string to an array
+    if (data.trim().startsWith("[") && data.trim().endsWith("]")) {
+      // Support for JSON arrays: ["foo", 1, null, "test"]
+      try {
+        return JSON.parse(data)
+      } catch (error) {
+        return [data]
+      }
+    } else {
+      // Support for comma-separated values: "foo,1,,test"
+      return data.split(",")
+    }
+  }
+
+  try {
+    const parsedData = JSON.parse(
+      JSON.stringify(data, (_key, value) =>
+        typeof value === "bigint" ? Number(value) : value
+      )
+    )
+    if (!isArray(parsedData)) {
+      return [toSafeString(parsedData)]
+    }
+
+    return parsedData.map((value: any) =>
+      ["string", "number", "boolean", "null"].includes(typeof value)
+        ? value
+        : toSafeString(value)
+    )
+  } catch (error) {
+    return [toSafeString(data)]
+  }
+}
+
+/**
+ * Converts the given value of unknown type to a string without
+ * the risks of any exceptions.
+ *
+ * @param data - The value to convert to a string.
+ *
+ * @return The converted string or a string showing the type of the object as fallback.
+ */
+export function toSafeString(data: any): string {
+  try {
+    try {
+      return toString(data)
+    } catch (error) {
+      return JSON.stringify(data, (_key, value) =>
+        typeof value === "bigint" ? Number(value) : value
+      )
+    }
+  } catch (error) {
+    return `[${typeof data}]`
```

Comment:
should this return the error instead of the type of data?

Reply:
This is basically how javascript console.log also deals with objects that it cannot convert to a string: It prints out: `[object Object]`. Since the function should explicitly not return any error or error message, I would not put any error message here. The type in the brackets are the best we can do to show information about the underlying value.